compiler/src/dotty/tools/dotc/core/Decorators.scala

```diff
@@ -112,9 +112,11 @@ object Decorators {
         else x1 :: xs1
       }
 
-    def foldRightBN[U](z: => U)(op: (T, => U) => U): U = xs match {
+    def foldRightBN[U](z: => U)(op: (T, => U) => U): U = xs.reverse.foldLeftBN(z)(op)
+
+    @tailrec final def foldLeftBN[U](z: => U)(op: (T, => U) => U): U = xs match {
```

Comment:
`foldLeft` in the standard library has the accumulator as the first arg of the closure:
```scala
def foldLeft[U](z: U)(op: (U, T) => U): U
```

Reply:
Oh yes, `z` is supposed to be called `acc`

Reply:
I will create a separate PR with this.